[Inductor][Triton] Support TMA before strict 3.4 cutoff

njriasan opened the pull request:
Summary: Inductor's 3.4 Triton release is the most common used variant of Triton, but if someone is working with an alternative version of Triton this may not match. This moves the version check from 3.4 Triton to any variant that has support for the TMA APIs.

Test Plan:
Relying on CI. Should be a NFC.

Rollback Plan:

Reviewed By: davidberard98

Differential Revision: D79378792


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

davidberard98 reviewed (approved):
it needs to be linted, otherwise lgtm & thanks for the fix!

njriasan committed:
[Inductor][Triton] Support TMA before strict 3.4 cutoff (#159777)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/159777

Inductor's 3.4 Triton release is the most common used variant of Triton, but if someone is working with an alternative version of Triton this may not match. This moves the version check from 3.4 Triton to any variant that has support for the TMA APIs.

Test Plan:
Relying on CI. Should be a NFC.

Rollback Plan:

Reviewed By: davidberard98

Differential Revision: D79378792

izaitsevfb commented:
@pytorchbot revert -m "breaking inductor test on ROCm" -c nosignal

```
FAILED [2.6870s] inductor/test_torchinductor_strided_blocks.py::TritonTensorDescriptorTestCUDA::test_welford_non_block_pointer_cuda - AssertionError: Scalars are not equal!
caused by this PR:
https://github.com/pytorch/pytorch/pull/159777
[Inductor][Triton] Support TMA before strict 3.4 cutoff #159777
```

hud link:
https://hud.pytorch.org/hud/pytorch/pytorch/main/1?per_page=50&name_filter=rocm%20%2F&mergeEphemeralLF=true

njriasan commented:
@pytorchbot merge
